tests/tracer/test_logging.py

```diff
@@ -0,0 +1,284 @@
+        the child logger inherits from ddtrace automatically.
+    """
+    env = os.environ.copy()
+    env["DD_TRACE_DEBUG"] = "true"
+    log_file = tmpdir + "/testlog.log"
+    env["DD_TRACE_LOG_FILE"] = log_file
+    code = """
+import logging
+import ddtrace
+
+ddtrace.patch_all()
+
+ddtrace_logger = logging.getLogger('ddtrace')
+assert ddtrace_logger.level == logging.DEBUG
+assert ddtrace_logger.handlers[0].__class__.__name__ == 'RotatingFileHandler'
+assert ddtrace_logger.handlers[0].maxBytes == 15 << 20
+
+child_logger = logging.getLogger('ddtrace.child')
+assert child_logger.parent.name == 'ddtrace'
+assert child_logger.getEffectiveLevel() == logging.DEBUG
+assert child_logger.handlers == []
+"""
+
+    out, err, status, pid = run_python_code_in_subprocess(code, env=env)
+    assert status == 0, err
+    assert err == b""
+    assert out == b""
+    with open(log_file) as file:
+        content = file.read()
+        assert len(content) > 0
+
+
+def test_debug_logs_go_to_stderr_patch_all(run_python_code_in_subprocess):
+    """
+    When setting up the default logger in debug mode,
+        it automatically logs to stderr.
+    """
+    env = os.environ.copy()
+    env["DD_TRACE_DEBUG"] = "true"
+    code = """
+import logging
+import ddtrace
+
+ddtrace.patch_all()
+
+ddtrace_logger = logging.getLogger('ddtrace')
+assert ddtrace_logger.level == logging.DEBUG
+assert ddtrace_logger.handlers[0].__class__.__name__ == 'StreamHandler'
+"""
+
+    out, err, status, pid = run_python_code_in_subprocess(code, env=env)
+    assert status == 0, err
+    assert err != b""
+    assert out == b""
+
+
+def test_warn_logs_go_to_stderr_patch_all(run_python_code_in_subprocess):
+    """
+    When setting up the default logger,
+        it automatically logs to stderr at WARN log level.
+    """
+    code = """
+import logging
+import ddtrace
+
+ddtrace.patch_all()
+
+ddtrace_logger = logging.getLogger('ddtrace')
+assert ddtrace_logger.level == logging.WARN
+assert ddtrace_logger.handlers[0].__class__.__name__ == 'StreamHandler'
+
+ddtrace_logger.warning('warning log')
+"""
+
+    out, err, status, pid = run_python_code_in_subprocess(code)
+    assert status == 0, err
+    assert err != b""
+    assert out == b""
+
+
+def test_debug_logs_go_to_stderr_ddtrace_run(ddtrace_run_python_code_in_subprocess):
+    """
+    When setting up the default logger with ddtrace-run in debug mode,
+        it automatically logs to stderr.
+    """
+    env = os.environ.copy()
+    env["DD_TRACE_DEBUG"] = "true"
+
+    code = """
+import logging
+from ddtrace.logger import configure_ddtrace_logger
+import ddtrace
+
+ddtrace_logger = logging.getLogger('ddtrace')
+assert ddtrace_logger.level == logging.DEBUG
+assert ddtrace_logger.handlers[0].__class__.__name__ == 'StreamHandler'
+"""
+    out, err, status, pid = ddtrace_run_python_code_in_subprocess(code, env=env)
+    assert status == 0, err
+    assert err != b""
```

Comment:
this is a good example of where asserting the content might be a good idea, since `configure_ddtrace_logger` is the thing logging (we don't have an explicit log in the test scenario)

Reply:
I'm going to mark this as resolved since in the latest version, there has been commits to assert on the content.